apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/account/service/AccountPublicKeyServiceImpl.java

```diff
@@ -154,13 +154,29 @@ public PublicKey loadPublicKey(DbKey dbKey) {
 
     @Override
     public PublicKey loadPublicKey(DbKey dbKey, int height) {
-        PublicKey publicKey = publicKeyTable.get(dbKey, height);
+        PublicKey publicKey = getPublicKey(dbKey, height);
         if (publicKey == null) {
-            publicKey = genesisPublicKeyTable.get(dbKey, height);
+            publicKey = getGenesisPublicKey(dbKey, height);
         }
         return publicKey;
     }
 
+    private PublicKey getGenesisPublicKey(DbKey dbKey, int height) {
+        if (height < 0 || blockChainInfoService.doesNotExceed(height)) {
```

Comment:
genesis public keys are unchangeable, so that such checks are useless

Reply:
It was done earlier for all entities, so transfer all the logic. Taking into account your point, it may be removed